sdk/core/ts-http-runtime/src/fetchHttpClient.ts

```diff
@@ -221,27 +221,44 @@ function buildPipelineHeaders(httpResponse: Response): PipelineHeaders {
 }
 
 interface BuildRequestBodyResponse {
-  body:
-    | string
-    | Blob
-    | ReadableStream<Uint8Array>
-    | ArrayBuffer
-    | ArrayBufferView
-    | FormData
-    | null
-    | undefined;
+  body?: BodyInit;
   streaming: boolean;
 }
 
+export function arrayBufferViewToArrayBuffer(source: ArrayBufferView): ArrayBuffer {
+  if (
+    source.buffer instanceof ArrayBuffer &&
+    source.byteOffset === 0 &&
+    source.byteLength === source.buffer.byteLength
+  ) {
+    return source.buffer;
+  }
+
+  const arrayBuffer = new ArrayBuffer(source.byteLength);
```

Comment:
maybe we don't need to create a copy if the ArrayBufferView is used only for input and nothing else?